addOns/oast/src/main/java/org/zaproxy/addon/oast/scripts/ExtensionOastScripts.java

```diff
@@ -0,0 +1,177 @@
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.zaproxy.addon.oast.ExtensionOast;
+import org.zaproxy.zap.extension.graaljs.ExtensionGraalJs;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptEngineWrapper;
+import org.zaproxy.zap.extension.script.ScriptType;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+
+public class ExtensionOastScripts extends ExtensionAdaptor {
+
+    private static final List<Class<? extends Extension>> DEPENDENCIES;
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionOastScripts.class);
+    private static final String TEMPLATE_REGISTER_REQUEST_HANDLER =
+            "OAST Register Request Handler.js";
+    private static final String SCRIPT_GET_BOAST_SERVERS = "OAST Get BOAST Servers.js";
+
+    private ExtensionScript extScript;
+
+    static {
+        List<Class<? extends Extension>> dependencies = new ArrayList<>(3);
+        dependencies.add(ExtensionOast.class);
+        dependencies.add(ExtensionScript.class);
+        dependencies.add(ExtensionGraalJs.class);
+        DEPENDENCIES = Collections.unmodifiableList(dependencies);
+    }
+
+    @Override
+    public String getName() {
+        return ExtensionOastScripts.class.getSimpleName();
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("oast.scripts.desc");
+    }
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
+
+    @Override
+    public void postInit() {
+        extScript = Control.getSingleton().getExtensionLoader().getExtension(ExtensionScript.class);
+        addScripts();
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        removeScripts();
+    }
+
+    private void addScripts() {
+        if (extScript == null) {
+            LOGGER.warn(Constant.messages.getString("oast.scripts.warn.couldNotAddScripts"));
+            return;
+        }
+        addScript(
+                TEMPLATE_REGISTER_REQUEST_HANDLER,
+                Constant.messages.getString("oast.scripts.requestHandler.desc"),
+                extScript.getScriptType(ExtensionScript.TYPE_STANDALONE),
+                true);
+        addScript(
+                SCRIPT_GET_BOAST_SERVERS,
+                Constant.messages.getString("oast.scripts.getBoastServers.desc"),
+                extScript.getScriptType(ExtensionScript.TYPE_STANDALONE),
+                false);
+    }
+
+    private void addScript(String name, String description, ScriptType type, boolean isTemplate) {
+        try {
+            if (extScript == null || extScript.getScript(name) != null) {
```

Comment:
The `extScript` was validated earlier, although I don't think it can be null (if it's disabled this extension will be as well).